cloudinit/sources/DataSourceCloudCIX.py

```diff
@@ -0,0 +1,206 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import json
+import logging
+from typing import Optional
+
+from cloudinit import dmi, net, sources, url_helper, util
+from cloudinit.net.dhcp import NoDHCPLeaseError
+from cloudinit.net.ephemeral import EphemeralIPNetwork
+
+LOG = logging.getLogger(__name__)
+
+METADATA_URLS = ["http://169.254.169.254"]
+METADATA_VERSION = 1
+
+BUILTIN_DS_CONFIG = {
+    "retries": 3,
+    "timeout": 5,
+    "wait": 5,
+}
+
+
+CLOUDCIX_DMI_NAME = "CloudCIX"
+
+
+class DataSourceCloudCIX(sources.DataSource):
+
+    dsname = "CloudCIX"
```

Comment:
Let's define the url_*params as class attributes and use the get_url_params()  method to extract this config from ds_cfg for us.

```suggestion
    dsname = "CloudCIX"
    
    # Setup read_url parameters per get_url_params.
    url_retries = 3
    url_timeout = 5
    url_sec_between_retries = 5
```

Reply:
Done.